arch/arm/core/cortex_m/mpu/arm_mpu.c

```diff
@@ -78,251 +126,138 @@ void arm_core_mpu_disable(void)
 	MPU->CTRL = 0;
 }
 
-#if defined(CONFIG_USERSPACE) || defined(CONFIG_MPU_STACK_GUARD) || \
-	defined(CONFIG_APPLICATION_MEMORY) || defined(CONFIG_NOCACHE_MEMORY)
-
+#if defined(CONFIG_USERSPACE)
 /**
- * This internal function is utilized by the MPU driver to parse the intent
- * type (i.e. THREAD_STACK_REGION) and to fill-in the correct parameter
- * set to the provided attribute structure.
- *
- * @return 0 on Success, otherwise return -EINVAL.
+ * @brief configure an active memory partition
  */
-static inline int _get_region_attr_by_type(arm_mpu_region_attr_t *p_attr,
-	u32_t type, u32_t base, u32_t size)
+void arm_core_mpu_mem_partition_configure(struct k_mem_partition *partition,
```

Comment:
..._static_region_update()?

Reply:
This is _not_ for static regions - memory partitions are dynamic.
But, I think, I see your point; this function performs a _configuration update_ of an existing region.

I can add the _update in the function name if you insist, @pizi-nordic

Reply:
(addressed)

Reply:
Ok.